src/components/csvIntegration.js

```diff
@@ -0,0 +1,194 @@
+import { urlWriteAppendMultipleObjects } from "../urls";
+
+import style from "../styles.css";
+
+export class WriteAppendMultipleObjects extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modelName: upperFirst(props.componentName),
+      data: {
+        "csv": null,
+        "csvLink": null,
+        "uploadType": "",
+      },
+      errors: [],
+    };
+  }
+
+  componentDidMount() {
+    document.addEventListener("keydown", this.handleKeyPress, false);
+  }
+  componentWillUnmount() {
+    document.removeEventListener("keydown", this.handleKeyPress, false);
+  }
+
+  handleKeyPress = e => {
+    if (e.keyCode === 27) {
+      this.props.handleCsvHide(this.props.componentName);
+    }
+  };
+
+  downloadCsv = () => {
+    axios({
+      method: "get",
+      url: urlWriteAppendMultipleObjects(),
+      params: {
+        model: this.state.modelName
+      },
+    }).then((response) => {
+      const url = window.URL.createObjectURL(new Blob([response.data]));
+      const link = document.createElement("a");
+      const name = response.headers["content-disposition"].split(
+        "filename="
+      )[1];
+      link.href = url;
+      link.setAttribute("download", name);
+      document.body.appendChild(link);
+      link.click();
+    });
+  }
+
+  handleFile = (event, file, value, name) => {
+    let link = name + "Link";
+    this.setState({
+      data: {
+        ...this.state.data,
+        [name]: file,
+        [link]: value
+      }
+    });
+    event.target.value = "";
+  };
+
+  handleDelete = name => {
+    let link = name + "Link";
+    this.setState({
+      data: {
+        ...this.state.data,
+        [name]: null,
+        [link]: null
+      }
+    });
+  };
+
+  handleChange = (name, value) => {
+    this.setState({
+      data: {
+        ...this.state.data,
+        [value.name]: value.value
+      }
+    });
+  };
+
+  handleErrors = error_dict => {
+    let dict = error_dict;
+    let errors = [];
+    for (let key in dict) {
+      for (let index in dict[key]) {
+        errors.push(startCase(key) + ": " + dict[key][index]);
+      }
+    }
+    this.setState({ errors: errors });
+  };
+
+  handleSubmit = () => {
+    if (this.state.data.uploadType == "" || this.state.data.csvLink == null){
+      this.setState({errors: ["Please fill all the form fields"]})
+    } else {
+        let data = new FormData();
+        data.append("model", this.state.modelName)
+        for (let prop in this.state.data) {
+          data.append(snakeCase(prop), this.state.data[prop])
+        }
+        axios({
+            method: "post",
+            url: urlWriteAppendMultipleObjects(),
+            data: data,
+            headers: headers
+        }).then(response => {
+          this.props.handleUpdate(response.data, this.props.componentName);
+          this.props.handleCsvHide(this.props.componentName);
+        }).catch(error => {
+            if (error.response.status == "400") {
```

Comment:
Please google the difference between `==` and `===`. You will get why this line of code is so egregious.